src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -236,7 +236,7 @@ private CookieOptions BuildCookieOptions()
                 if (signInContext.Properties.IsPersistent)
                 {
                     var expiresUtc = signInContext.Properties.ExpiresUtc ?? issuedUtc.Add(Options.ExpireTimeSpan);
-                    signInContext.CookieOptions.Expires = expiresUtc.ToUniversalTime().DateTime;
+                    signInContext.CookieOptions.Expires = expiresUtc.LocalDateTime;
```

Comment:
@brentschmaltz regarding your confusion,
The `Kind` value of `expiresUtc.ToUniversalTime().DateTime` is `DateTimeKind.Unspecified` rather than `DateTimeKind.Utc` (Surprise!), this results in a bug in the following code:

```c#
// ChunkingCookieManager.cs, line 129
var template = new SetCookieHeaderValue(escapedKey)
{
    Domain = options.Domain,

    // there is an implicit conversion (from DateTime to DateTimeOffset) here.
    // options.Expires.Kind == DateTimeKind.Unspecified
    // hence options.Expires is treated as a local DateTime during the conversion while it's a UTC DateTime.
    Expires = options.Expires,

    HttpOnly = options.HttpOnly,
    Path = options.Path,
    Secure = options.Secure,
};
```